helper/schema/resource.go

```diff
@@ -151,7 +151,7 @@ func (r *Resource) Apply(
 		err = r.Update(data, meta)
 	}
 
-	return data.State(), err
+	return r.recordCurrentSchemaVersion(data.State()), err
```

Comment:
This was the bug that wasted me a nice little bit of time. :sob: See the first commit's msg for details.